packages/eslint-plugin/src/rules/type-annotation-spacing.ts

```diff
@@ -32,6 +47,107 @@ const definition = {
   additionalProperties: false,
 };
 
+function createRules(options?: Config): WhitespaceRules {
+  const globals = {
+    ...(options?.before !== undefined ? { before: options.before } : {}),
+    ...(options?.after !== undefined ? { after: options.after } : {}),
+  };
+  const override = options?.overrides ?? {};
+  const colon = {
+    ...{ before: false, after: true },
+    ...globals,
+    ...override?.colon,
+  };
+  const arrow = {
+    ...{ before: true, after: true },
+    ...globals,
+    ...override?.arrow,
+  };
+
+  return {
+    colon: colon,
+    arrow: arrow,
+    variable: { ...colon, ...override?.variable },
+    property: { ...colon, ...override?.property },
+    parameter: { ...colon, ...override?.parameter },
+    returnType: { ...colon, ...override?.returnType },
+  };
+}
+
+function getNodeType(node?: TSESTree.Node): string {
+  return node?.type ?? '';
+}
+
+function typeIsArrowFunction(type: string): boolean {
+  return type === AST_NODE_TYPES.TSFunctionType;
+}
+
+function typeIsIdentifier(type: string): boolean {
+  return type === AST_NODE_TYPES.Identifier;
+}
+
+function typeIsProperty(type: string): boolean {
+  return [
+    AST_NODE_TYPES.ClassProperty,
+    AST_NODE_TYPES.FunctionExpression,
+    AST_NODE_TYPES.TSPropertySignature,
+    AST_NODE_TYPES.TSMethodSignature,
+  ].includes(type as AST_NODE_TYPES);
+}
+
+function typeIsReturnType(type: string): boolean {
+  return [
+    AST_NODE_TYPES.FunctionDeclaration,
+    AST_NODE_TYPES.ArrowFunctionExpression,
+  ].includes(type as AST_NODE_TYPES);
+}
+
+function typeIsVariable(type: string): boolean {
+  return type === AST_NODE_TYPES.VariableDeclarator;
+}
+
+function typeIsParameter(type: string): boolean {
+  return [
+    AST_NODE_TYPES.FunctionDeclaration,
+    AST_NODE_TYPES.TSFunctionType,
+    AST_NODE_TYPES.TSMethodSignature,
+    AST_NODE_TYPES.TSEmptyBodyFunctionExpression,
+  ].includes(type as AST_NODE_TYPES);
+}
+
+function getIdentifierRules(
+  rules: WhitespaceRules,
+  node?: TSESTree.Node,
+): WhitespaceRule {
+  const scope = node?.parent;
+  const type = getNodeType(scope);
+
+  if (typeIsVariable(type)) {
+    return rules.variable;
+  } else if (typeIsParameter(type)) {
+    return rules.parameter;
+  } else {
+    return rules.colon;
+  }
+}
+
+function getRules(rules: WhitespaceRules, node: TSESTree.Node): WhitespaceRule {
```

Comment:
nit - you can use the correct type here
```suggestion
function getRules(rules: WhitespaceRules, node: TSESTree.TypeNode): WhitespaceRule {
```

Reply:
fair enough